tests/js/client/aql/aql-object-enumeration.js

```diff
@@ -258,13 +279,76 @@ function enumerateObjectTestSuite() {
 
     testOutputDocumentCollectionWithLimit4: function () {
       // skip 400 first values (therefore all documents will be skipped), return 4.
-      let actual = queryDocumentsWithLimit(docCollection, 400, 4, true);
+      let actual = queryDocumentsWithLimit(docCollection, 400, 4, true, ["+replace-entries-with-object-iteration"]);
 
       // Check fullCount value
       assertEqual(actual.getExtra()["stats"]["fullCount"], db[docCollection].count() * 4);
 
       actual = actual.toArray();
       assertTrue(actual.length === 0);
+    },
+
+    testOptimizationFilter: function () {
+      const query = `
+      FOR doc IN ${docCollection}
+        FOR [key, value] IN ENTRIES(doc)
+        FILTER key == 'key1' and value == 'a'
+        return [key, value]
+      `;
+  
+      let actual_optimized = queryDocuments(docCollection, query, /*fullCount=*/true, ["+replace-entries-with-object-iteration"]);
+      let actual_not_optimized = queryDocuments(docCollection, query, /*fullCount=*/true, ["-replace-entries-with-object-iteration"]);
```

Comment:
In general I would prefer const where possible
```suggestion
      const actual_optimized = queryDocuments(docCollection, query, /*fullCount=*/true, ["+replace-entries-with-object-iteration"]);
      const actual_not_optimized = queryDocuments(docCollection, query, /*fullCount=*/true, ["-replace-entries-with-object-iteration"]);
```